SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -174,12 +195,26 @@ private async void MapMouseLeftButtonDown(object sender, MouseButtonEventArgs e)
                 SYLMap.Center = MainMap.ViewToLocation(e.GetPosition(MainMap));
                 SYLMap.AddMarkerTemp(SYLMap.Center);
 
-                if (GridSellerAdd.Visibility != Visibility.Visible)
+                if (GridSellerAdd.Visibility == Visibility.Collapsed)
                 {
-                    SYLMap.DrawRadiusOnTempMarker(Slider1Seller.Value * 1000.0);
-                    var address = await MapUtilityProvider.LocationToAddress(SYLMap.Center.Latitude, SYLMap.Center.Longitude);
-                    TextBox2Seller.Text = address.Item2;
-                    TextBox3Seller.Text = address.Item1;
+                    if (GridSellersSearch.Visibility == Visibility.Visible)
+                    {
+                        SYLMap.DrawRadiusOnTempMarker(Slider1Seller.Value * 1000.0);
+                    }
+
+                    var address = SYLMap.LocationToAddressSplit(SYLMap.Center);
```

Comment:
Use the MapUtilityProvider function instead of the SYLMap one.
`var address = await MapUtilityProvider.LocationToAddress(SYLMap.Center.Latitude, SYLMap.Center.Longitude);`